src/main/java/io/nats/client/impl/NatsMessage.java

```diff
@@ -13,195 +13,159 @@
 
 package io.nats.client.impl;
 
-import java.nio.ByteBuffer;
-import java.nio.CharBuffer;
-import java.nio.charset.Charset;
-import java.nio.charset.StandardCharsets;
-
 import io.nats.client.Connection;
 import io.nats.client.Message;
 import io.nats.client.Subscription;
 
+import static java.nio.charset.StandardCharsets.US_ASCII;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
 class NatsMessage implements Message {
-    private String sid;
+    private static byte[] PUB_BYTES = "PUB ".getBytes(US_ASCII);

```

Comment:
imo the protocol bytes and lengths should be defined in NatsConnection with the others.

Reply:
I think they should be pulled out into a common constants file. I do not like getting them from NatsConnection because it makes a circular dependency, which I know the compiler handles fine. I'll fix stuff like this on the next PR